public_html/core/engine/dispatcher.php

```diff
@@ -286,86 +271,95 @@ public function dispatch($parent_controller = '')
             $error = new AError('Error: controller class not exist '.$this->class.'!', AC_ERR_CLASS_CLASS_NOT_EXIST);
             $error->toLog()->toDebug();
         }
-        if (is_callable([$controller, $this->method])) {
-            /**
-             * @var $dispatch ADispatcher
-             */
-            $r = new ReflectionMethod($controller, $this->method);
-            $params = $r->getParameters();
-            if(!$params){
-                $args = [];
-            }else {
-                $args = $this->args;
-            }
-
-            $dispatch = call_user_func_array([$controller, $this->method], $args);
-            //Check if return is a dispatch and need to call new page
-            if ($dispatch && is_object($dispatch)) {
-                if ($this->args["instance_id"] == 0) {
-                    //If main controller come back for new dispatch
-                    return $dispatch->getController().'/'.$dispatch->getMethod();
+        try {
+            if (is_callable([$controller, $this->method])) {
+                /**
+                 * @var $dispatch ADispatcher
+                 */
+                $r = new ReflectionMethod($controller, $this->method);
```

Comment:
What is ReflectionMethod?

Reply:
https://www.php.net/manual/en/class.reflectionmethod.php